token/cli/src/main.rs

```diff
@@ -3712,8 +3713,8 @@ async fn process_command<'a>(
 
             let (authority_signer, authority) =
                 config.signer_or_default(arg_matches, "authority", &mut wallet_manager);
-            if !bulk_signers.contains(&authority_signer) {
-                bulk_signers.push(authority_signer);
+            if config.multisigner_pubkeys.is_empty() {
+                push_signer_with_dedup(authority_signer, &mut bulk_signers);
```

Comment:
this is basically just a santity check right? if `multisigner_pubkeys` has a length then its already in the array anyway so dont bother to check?

Reply:
Oh yeah that's true. Since fee payer and nonce authority are stored separately, we don't actually need to do the dedup check. I'll remove it and just do a normal push.

Reply:
Oh nvm, I misunderstood your point. This `multisigner_pubkeys` check is crucial! If `multisigner_pubkeys` has a length, then we're in "multisig mode", so we don't want the authority to sign, since that's actually the multisig address, and that never signs. If it's added, then we'll create a transaction where it's a signer, which is incorrect.

But in "non-multisig mode", we need the authority to sign, so we must add it here. And just to be totally safe, I'll keep this as is with the dedup, just in case there's some bizarre edge case that we haven't considered.

Reply:
ohhh right, gotcha